src/generate_conf.rs

```diff
@@ -524,4 +539,51 @@ mod tests {
             Some("node1.example.com".as_ref())
         );
     }
+
+    #[test]
+    fn test_populate_connection_ids() -> Result<(), anyhow::Error> {
+        let exp_output_path = Path::new("testdata/generate/expected");
+        let mut exp_hosts: Vec<Host> = serde_yaml::from_str(
+            fs::read_to_string(exp_output_path.join(HOST_MAPPING_FILE))?.as_str(),
+        )?;
+        let exp_ifaces = exp_hosts.pop().unwrap().interfaces;
+        let mut ifaces: Vec<Interface> = exp_ifaces.clone();
+        ifaces.iter_mut().for_each(|i| {
+            i.connection_ids = Vec::new();
+        });
```

Comment:
I believe this shouldn't be necessary and then there's no point in cloning it either. WDYT?